snapcraft/cli/lifecycle.py

```diff
@@ -0,0 +1,209 @@
+        'Edit the file to your liking or run `snapcraft` to get started')
+
+
+@lifecyclecli.command()
+@click.pass_context
+@add_build_options()
+@click.argument('parts', nargs=-1, metavar='<part>...', required=False)
+def pull(ctx, parts, **kwargs):
+    """Download or retrieve artifacts defined for a part.
+
+    \b
+    Examples:
+        snapcraft pull
+        snapcraft pull my-part1 my-part2
+
+    """
+    _execute('pull', parts, **kwargs)
+
+
+@lifecyclecli.command()
+@add_build_options()
+@click.argument('parts', nargs=-1, metavar='<part>...', required=False)
+def build(parts, **kwargs):
+    """Build artifacts defined for a part.
+
+    \b
+    Examples:
+        snapcraft build
+        snapcraft build my-part1 my-part2
+
+    """
+    _execute('build', parts, **kwargs)
+
+
+@lifecyclecli.command()
+@add_build_options()
+@click.argument('parts', nargs=-1, metavar='<part>...', required=False)
+def stage(parts, **kwargs):
+    """Stage the part's built artifacts into the common staging area.
+
+    \b
+    Examples:
+        snapcraft stage
+        snapcraft stage my-part1 my-part2
+
+    """
+    _execute('stage', parts, **kwargs)
+
+
+@lifecyclecli.command()
+@add_build_options()
+@click.argument('parts', nargs=-1, metavar='<part>...', required=False)
+def prime(parts, **kwargs):
+    """Final copy and preparation for the snap.
+
+    \b
+    Examples:
+        snapcraft prime
+        snapcraft prime my-part1 my-part2
+
+    """
+    _execute('prime', parts, **kwargs)
+
+
+@lifecyclecli.command()
+@add_build_options()
+@click.argument('directory', required=False)
+@click.option('--output', '-o', help='path to the resulting snap.')
+def snap(directory, output, **kwargs):
+    """Create a snap.
+
+    \b
+    Examples:
+        snapcraft snap
+        snapcraft snap --output renamed-snap.snap
+
+    If you want to snap a directory, you should use the snap-dir command
+    instead.
+    """
+    project_options = get_project_options(**kwargs)
+    if env.is_containerbuild():
+        lifecycle.containerbuild('snap', project_options, output, directory)
+    else:
+        snap_name = lifecycle.snap(
+            project_options, directory=directory, output=output)
+        click.echo('Snapped {}'.format(snap_name))
+
+
+@lifecyclecli.command()
+@add_build_options()
+@click.argument('parts', nargs=-1, metavar='<part>...', required=False)
+@click.option('--step',
+              type=click.Choice(['pull', 'build', 'stage', 'prime', 'strip']),
+              help='only clean the specified step and those that '
+                   'depend on it.')
+def clean(parts, step, **kwargs):
+    """Remove content - cleans downloads, builds or install artifacts.
+
+    \b
+    Examples:
+        snapcraft clean
+        snapcraft clean my-part --step build
+    """
+    project_options = get_project_options(**kwargs)
+    if env.is_containerbuild():
+        step = step or 'pull'
+        lifecycle.containerbuild('clean', project_options,
```

Comment:
Might be nice to unify this more so that containerbuild and other lifecycle commands mirror each other. But it probably makes sense to attempt that in a follow-up as it's complex enough as it is.